objects/src/assets/vault.rs

```diff
@@ -89,6 +92,48 @@ impl AssetVault {
     // PUBLIC MODIFIERS
     // --------------------------------------------------------------------------------------------
 
+    pub fn apply_delta(&mut self, delta: &AccountVaultDelta) -> Result<(), AssetVaultError> {
+        for (&faucet_id, &delta) in delta.fungible() {
+            let asset = FungibleAsset::new(faucet_id, delta.unsigned_abs())
+                .map_err(AssetVaultError::FungibleAssetValidationError)?;
+            if delta >= 0 {
+                self.add_fungible_asset(asset)?;
+            } else {
+                self.remove_fungible_asset(asset)?;
+            }
+        }
+
+        for (&asset, &action) in delta.non_fungible() {
+            // # SAFETY: the asset must be a valid word representation of a non-fungible asset
```

Comment:
I don't see another good solution, because we use `add` method of fungible asset in order to update its amount. We could change only processing of non-fungible assets, but I don't think it would be a good idea.